src/cascade/input_data/configuration/builder.py

```diff
@@ -26,6 +58,218 @@ def initial_context_from_epiviz(configuration):
     return context
 
 
+def unique_country_covariate_transform(configuration):
+    """
+    Iterates through all covariate IDs, including the list of ways to
+    transform them, because each transformation is its own column for Dismod.
+    """
+    seen_covariate = defaultdict(set)
+    if configuration.country_covariate:
+        for covariate_configuration in configuration.country_covariate:
+            seen_covariate[covariate_configuration.country_covariate_id].add(covariate_configuration.transformation)
+
+    for cov_id, cov_transformations in seen_covariate.items():
+        yield cov_id, list(sorted(cov_transformations))
+
+
+def assign_covariates(model_context, configuration):
+    """
+    The EpiViz interface allows assigning a covariate with a transformation
+    to a specific target (rate, measure value, measure standard deviation).
+    It is even the case, that the same covariate, say income, can be applied
+    without transformation to iota on one smoothing and applied without
+    transformation to chi with a *different smoothing*.
+    Therefore, there can be multiple covariate columns built from the same
+    covariate, one for each kind of transformation required.
```

Comment:
For income, the previous sentence talks about no transformation for the iota case, and no transformation for the chi case.  The last sentence suggests that two different transformations were used and two columns created, but "no transformation" = "no transformation", so just wondering.